GHA CI and PHP 8

gsteel opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

This pull add the GHA CI setup, removing travis config.
It also refactors tests similar to #5 with ref to #4 but that pull looks stuck

The reason for the pull is to hopefully get it merged so that I can tackle adding Psalm and possibly update the coding standard to ^2

gsteel committed:
Bump minimum required PHP version to 7.3 and allow 8.0

Upgrade to recent version of PHPUnit

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Fix setUp() method signature in all test cases

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Refactor tests
- Minimise prophecy usage in favour of PHPUnit mocks
- Add the Prophecy PHPUnit trait to reinstate prophecy support
- Fix deprecated or removed PHPUnit assertions

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Remove travis config, add GHA config, add composer lock file

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Update README.md with GHA status badge

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Migrate PHPUnit configuration

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
laminas-view at 2.12.0 minimum is required to support PHP 8.0

Signed-off-by: George Steel <george@net-glue.co.uk>

froschdesign commented on composer.json:
```diff
@@ -28,9 +28,9 @@
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.3||8.0",
```
PHP does not follow semantic versioning, therefore:
```suggestion
        "php": "^7.3 || ~8.0.0",
```

froschdesign commented on .github/workflows/continuous-integration.yml:
```diff
@@ -0,0 +1,33 @@
+      - name: ${{ matrix.name }}
+        uses: laminas/laminas-continuous-integration-action@v1
+        with:
+          job: ${{ matrix.job }}
```
This small error should be fixed. Thanks!

gsteel committed:
qa: Change PHP version constraint to allow patch and minor release. Update lock file. Add new line to GHA CI Yaml

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel commented:
Thank you for the review @froschdesign - I have also rebased on 3.x

froschdesign commented on composer.json:
```diff
@@ -28,9 +28,9 @@
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.3 || ~8.0",
```
Be careful here:

> The ~ operator is best explained by example: `~1.2` is equivalent to `>=1.2 <2.0.0`, while `~1.2.3` is equivalent to `>=1.2.3 <1.3.0`.

https://getcomposer.org/doc/articles/versions.md#tilde-version-range-

```suggestion
        "php": "^7.3 || ~8.0.0",
```

gsteel commented:
Leaving a polite nag for review from perhaps @froschdesign or @weierophinney … Sorry … I don't know who the maintainers are for this package 😬

weierophinney committed:
docs: removes CHANGELOG.md

Changelogs will be kept in milestone descriptions and propagated to tag and release notes.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: ignore PHPUnit result cache file

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
